Exercise_9/entry.js

```diff
@@ -0,0 +1,26 @@
+import MyFramework from './src/MyFramework';
+
+const BASIC_URL = 'http://localhost:8080';
+const TEMPLATES_URL = 'http://localhost:8080/templates';
+const WD = new MyFramework();
+const APP_NAME = 'app';
+/* global window:true */
+window.onload = () => {
+  WD.createRoot(APP_NAME)
+    .routes([
+      { url: `${BASIC_URL}`, templateUrl: `${TEMPLATES_URL}/home.html` },
+      { url: `${BASIC_URL}/sign-in.html`, templateUrl: `${TEMPLATES_URL}/sign-in.html` }]);
+
+  WD.root(APP_NAME).navigate(`${BASIC_URL}`);
```

Comment:
If your current location is "sign-in" you still will be redirected to the root. If should check at the beginning that you are located on the existing URL.